pkg/istio-agent/agent.go

```diff
@@ -403,14 +403,36 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 		return nil, fmt.Errorf("failed to start local DNS server: %v", err)
 	}
 
-	a.secretCache, err = a.newSecretManager()
-	if err != nil {
-		return nil, fmt.Errorf("failed to start workload secret manager %v", err)
+	sdsSocketExists := socketFileExists(constants.WorkloadIdentitySocketPath)
```

Comment:
I've been looking a lot at this section and wondering if they're not gonna complain about `Agent.Run()` getting too large, or ask us to flatten the logic (to avoid stacking if/elses inside of each other) but also having a hard time figuring out how improve on these matters 😅